docs/_future/create_new_app.md

```diff
@@ -0,0 +1,765 @@
+
+> 💡 In order to run the trigger and launch the full workflow, your application must be running via `slack run`.
+
+To learn more about Triggers and create new ones, visit the guide [here](/bolt-js/future/triggers).
+
+---
+### Functions {#functions}
+
+Functions are building blocks of automation that accept inputs, perform some calculations, and provide outputs. Functions can be used as steps in [Workflows](/bolt-js/future/workflows).
+
+There are two types of functions: [built-in functions](/bolt-js/future/built-in-functions), which are Slack native actions defined in the Schema, such as sending messages, and [custom functions](/bolt-js/future/custom-functions), which are not built-in and are defined by the developer.
+
+In the Request Time Off application, we use one custom function, `ApprovalFunction`, which we passed into our Time Off Workflow as the second step after a requester submits time off information. At this point, our custom function will take that information and then send it to the manager to approve or deny.
+
+Our `ApprovalFunction` is defined in `manifest/functions/approval.js`. Custom functions are defined using `DefineFunction`.
+```js
+// manifest/functions/approval.js
+const { DefineFunction, Schema } = require('@slack/bolt');
+
+const ApprovalFunction = DefineFunction({
+  callback_id: 'review_approval',
+  title: 'Approval Function',
+  description: 'Get approval for a request',
+  input_parameters: {
+    properties: {
+      employee: {
+        type: Schema.slack.types.user_id,
+        description: 'Requester',
+      },
+      manager: {
+        type: Schema.slack.types.user_id,
+        description: 'Manager',
+      },
+      start_date: {
+        type: Schema.slack.types.timestamp,
+        description: 'Start Date',
+      },
+      end_date: {
+        type: Schema.slack.types.timestamp,
+        description: 'End Date',
+      },
+    },
+    required: [
+      'employee',
+      'manager',
+      'start_date',
+      'end_date',
+    ],
+  },
+  output_parameters: {
+    properties: {},
+    required: [],
+  },
+});
+
+module.exports = { ApprovalFunction };
+```
+Within `ApprovalFunction`'s definition, we are declaring expected input and output parameters. We are also adding some additional metadata about our function, like a `title` and `description`.
+
+Parameters for the function are typed using [types from Slack's schema](https://api.slack.com/future/types), such as having the `user_id` type for the `employee` parameter.
+
+Once the function is defined, we need to create a listener for it to perform any desired functionality. In this case, our desired functionality is to send the inputted manager the time off request as a message.
+
+The function listener is declared in `listeners/functions/request-approval.js`:
```

Comment:
The code sample below is on the longer side, should we remove it and link to the file instead for easier maintainability?

Reply:
sounds good to me!